Modules/xSharePoint/DSCResources/MSFT_xSPAddInManagementServiceApp/MSFT_xSPAddInManagementServiceApp.psm1

```diff
@@ -0,0 +1,104 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]  [System.String] $Name,
+        [parameter(Mandatory = $true)]  [System.String] $ApplicationPool,
+        [parameter(Mandatory = $false)] [System.String] $DatabaseName,
+        [parameter(Mandatory = $false)] [System.String] $DatabaseServer,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+    Write-Verbose -Message "Getting Add-in service app '$Name'"
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        $serviceApps = Get-SPServiceApplication -Name $params.Name -ErrorAction SilentlyContinue 
+        if ($null -eq $serviceApps) { 
+            return $null 
+        }
+        $serviceApp = $serviceApps | Where-Object { $_.TypeName -eq "App Management Service Application" }
+
+        If ($null -eq $serviceApp) { 
+            return $null 
+        } else {
+            $returnVal =  @{
+                Name = $serviceApp.DisplayName
+                ApplicationPool = $serviceApp.ApplicationPool.Name
+                DatabaseName = $serviceApp.Database.Name
+                DatabaseServer = $serviceApp.Database.Server.Name
+                InstallAccount = $params.InstallAccount
+            }
+            return $returnVal
+        }
+    }
+    return $result
+}
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)]  [System.String] $Name,
+        [parameter(Mandatory = $true)]  [System.String] $ApplicationPool,
+        [parameter(Mandatory = $false)] [System.String] $DatabaseName,
+        [parameter(Mandatory = $false)] [System.String] $DatabaseServer,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    $result = Get-TargetResource @PSBoundParameters
+
+    if ($result.Count -eq 0) { 
```

Comment:
Can we check for $result -eq $null instead of count -eq 0 here seeing as we return null from the get method. I know we do this elsewhere but it'll get fixed when I get time to do a cleanup